### PR TITLE
refactor: unify context in webpack module

### PR DESCRIPTION
**Pull request by clarkdo**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/webpack/src/config/base.js [hunk 1]
```diff
@@ -43,17 +36,29 @@ export default class WebpackBaseConfig {
...
 
+  get mode() {
+    return this.dev ? 'development' : 'production'
+  }
+
+  get dev() {
+    return this.context.options.dev
+  }
+
+  get buildOpts() {
```
Comment (manniL) [inline packages/webpack/src/config/base.js]: Might be just personal taste but I'd prefer writing out `options` 🙈 

```suggestion
  get buildOptions() {
```

File: packages/webpack/src/config/base.js [hunk 2]
```diff
@@ -20,16 +20,9 @@ import WarnFixPlugin from '../plugins/warnfix'
...
-    this.isServer = options.isServer
-    this.isModern = options.isModern
+  constructor(builder) {
     this.builder = builder
-    this.nuxt = builder.context.nuxt
-    this.isStatic = builder.context.isStatic
-    this.options = builder.context.options
-    this.loaders = this.options.build.loaders
-    this.buildMode = this.options.dev ? 'development' : 'production'
+    this.context = builder.context
```
Comment (pi0) [inline packages/webpack/src/config/base.js]: `this.context` name is still ambiguous. Maybe we can call it `this.builderContext`?

File: packages/webpack/src/config/base.js (hunk 1 above)
Comment (pi0) [inline packages/webpack/src/config/base.js]: This is a REALLY common usage between classes. So why not moving getter to the context and use `this.buildContext.buildOptions`?

Comment (clarkdo) [reply packages/webpack/src/config/base.js]: It was, but I feel it's still to long.
OK， I'll move

File: packages/webpack/src/config/base.js (hunk 2 above)
Comment (clarkdo) [reply packages/webpack/src/config/base.js]: OK

File: packages/webpack/src/config/base.js (hunk 1 above)
Comment (clarkdo) [reply packages/webpack/src/config/base.js]: OK

**Review by galvez — DISMISSED** [review]
LGTM

File: packages/webpack/src/config/base.js [hunk 3]
```diff
@@ -142,13 +149,14 @@ export default class WebpackBaseConfig {
...
+    const { terser, cache } = this.buildContext.buildOptions
 
     // https://github.com/webpack-contrib/terser-webpack-plugin
-    if (this.options.build.terser) {
+    if (terser) {
       minimizer.push(
         new TerserWebpackPlugin(Object.assign({
           parallel: true,
-          cache: this.options.build.cache,
+          cache: cache,
```
Comment (manniL) [inline packages/webpack/src/config/base.js]:
```suggestion
          cache,
```

**Review by pi0 — APPROVED** [review]

**Pull request merged**